src/test/java/net/consensys/orion/http/ClientConnectionWhiteListTest.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.tuweni.concurrent.AsyncResult;
+import org.apache.tuweni.concurrent.CompletableAsyncResult;
+import org.apache.tuweni.junit.TempDirectory;
+import org.apache.tuweni.junit.TempDirectoryExtension;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(TempDirectoryExtension.class)
+class ClientConnectionWhiteListTest {
+
+  private final static Vertx vertx = vertx();
+  private static HttpClient httpClient;
+  private static HttpClient httpClientWithUnregisteredCert;
+  private static HttpClient httpClientWithImproperCertificate;
+  private static HttpClient httpClientWithMismatchedTrust;
+  private static int clientPort;
+  private static Orion orion;
+
+  @BeforeAll
+  static void setUp(@TempDirectory final Path tempDir) throws Exception {
+    final SelfSignedCertificate serverCertificate = SelfSignedCertificate.create("localhost");
+    final Config config = generateAndLoadConfiguration(tempDir, writer -> {
+      writer.write("clientconnectiontls='strict'\n");
+      writer.write("clientconnectiontlsservertrust='whitelist'\n");
+      writeServerCertToConfig(writer, serverCertificate);
+      writeClientConnectionServerCertToConfig(writer, serverCertificate);
+    });
+
+    configureJDKTrustStore(serverCertificate, tempDir);
+    final Path knownClientsFile = config.clientConnectionTlsKnownClients();
+
+    final SelfSignedCertificate clientCertificate = SelfSignedCertificate.create("example.com");
+    final String fingerprint = certificateHexFingerprint(Paths.get(clientCertificate.keyCertOptions().getCertPath()));
+    Files.write(knownClientsFile, Arrays.asList("#First line", "example.com " + fingerprint));
+    httpClient = vertx
+        .createHttpClient(new HttpClientOptions().setSsl(true).setKeyCertOptions(clientCertificate.keyCertOptions()));
+
+    final SelfSignedCertificate fooCertificate = SelfSignedCertificate.create("foo.bar.baz");
+    httpClientWithUnregisteredCert =
+        vertx.createHttpClient(new HttpClientOptions().setSsl(true).setKeyCertOptions(fooCertificate.keyCertOptions()));
+
+    final SelfSignedCertificate noCNCert = SelfSignedCertificate.create("");
+    httpClientWithImproperCertificate =
+        vertx.createHttpClient(new HttpClientOptions().setSsl(true).setKeyCertOptions(noCNCert.keyCertOptions()));
+
+    httpClientWithMismatchedTrust = vertx.createHttpClient(
+        new HttpClientOptions().setSsl(true).setKeyCertOptions(clientCertificate.keyCertOptions()).setTrustOptions(
+            new PemTrustOptions().addCertPath(clientCertificate.certificatePath())));
+
+    clientPort = config.clientPort();
+    orion = new Orion(vertx);
+    orion.run(System.out, System.err, config);
+  }
+
+  @AfterAll
+  static void tearDown() {
+    orion.stop();
+    vertx.close();
+  }
+
+  @Test
+  void clientInWhiteListCanInvokeUpcheck() throws Exception {
```

Comment:
nit: capital C in check i.e. upCheck and same below

Reply:
done